pkg/cvo/sync_worker.go

```diff
@@ -1096,6 +1096,25 @@ func (r *consistentReporter) ContextError(err error) error {
 	return errContext{fmt.Errorf("update %s at %d of %d", err, r.done, r.total)}
 }
 
+func (r *consistentReporter) CompleteWithErrors(errs []error) {
+	err := summarizeTaskGraphErrors(errs)
+
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	metricPayload.WithLabelValues(r.version, "pending").Set(float64(r.total - r.done))
```

Comment:
could possibly simplify with `consistentReporter.complete` and `consistentReporter.errors` helpers, and then have:

```go
func (r *consistentReporter) Complete() {
	r.lock.Lock()
	defer r.lock.Unlock()
	r.complete(r.status)
}

func (r *consistentReporter) Errors(errs []error) error {
	r.lock.Lock()
	defer r.lock.Unlock()
	_, err := r.errors(r.status, errs)
	return err
}
```

etc. to handle the locking.  Then this new function could be:

```go
func (r *consistentReporter) CompleteWithErrors(errs []error) {
	r.lock.Lock()
	defer r.lock.Unlock()
	copied, err := r.errors(r.status)
	copied = r.complete(copied)
}